training/metrics.py

```diff
@@ -24,92 +21,244 @@ def count_trainable_params(model):
       Total number ot trainable parameters
   """
   weights = model.trainable_weights
-  total_trainable_params = int(sum(np.prod(p.shape.as_list()) for p in object_identity.ObjectIdentitySet(weights)))
+  total_trainable_params = int(sum(np.prod(tf.size(p)) for p in object_identity.ObjectIdentitySet(weights)))
   return total_trainable_params
 
+def _count_linear_layer(layer, num_of_blades):
+  """
+  Note: This function calculates the flops for any linear layer
 
-def _count_flops_conv2d(layer):
-  if layer.data_format == "channels_first":
-    input_channels = layer.input_shape[1]
-    output_channels, h, w, = layer.output_shape[1:]
-  elif layer.data_format == "channels_last":
-    input_channels = layer.input_shape[3]
-    h, w, output_channels = layer.output_shape[1:]
-  w_h, w_w = layer.kernel_size
-
-  n_mult = h * w * output_channels * input_channels * w_h * w_w
-  n_add = w_h * w_w * input_channels * h * w * output_channels
+  Args:
+    layer (tf.keras.layer.Conv2D or DepthwiseConv2D or Dense): any linear layer from the keras model
+    num_of_blades: This value represent the number of blades we use for different upstride types.
 
-  flops = n_mult + n_add
+  Returns:
+      int : total FLOPs for the given layer.
+  """
+  input_shape = layer.input_shape
+  output_shape = layer.output_shape
+  output_group = np.prod(output_shape[1:])
+  class_name = layer.__class__.__name__
+
+  # we get output image size (H*W) (this is channel first/last dependent)
+  if len(input_shape) == 4: # 2D Conv and DepthWise
```

Comment:
For now we have linear layers which are always length 4. (B, C, H, W or B, H, W, C) that would satisfy this condition. Do you see a place where this would not work ?

Reply:
To me, we should make an if statement on the layer name, and then inside the if, make an assert on input_shape. That way we only capture the layers we explicitly want